docs/samples/Microsoft.ML.Samples/Dynamic/SimpleDataViewImplementation.cs

```diff
@@ -0,0 +1,258 @@
+                new InputObject(true, "Masterfully done hero!")
+            };
+            var dataView = new InputObjectDataView(inputArray);
+
+            // So, this is a very simple pipeline: a transformer that tokenizes Text, does nothing with the Label column
+            // at all.
+            var mlContext = new MLContext();
+            var transformedDataView = mlContext.Transforms.Text.TokenizeIntoWords(
+                "TokenizedText", "Text").Fit(dataView).Transform(dataView);
+
+            var textColumn = transformedDataView.Schema["Text"];
+            var tokensColumn = transformedDataView.Schema["TokenizedText"];
+
+            using (var cursor = transformedDataView.GetRowCursor(new[] { textColumn, tokensColumn }))
+            {
+                // Note that it is best to get the getters and values *before* iteration, so as to faciliate buffer
+                // sharing (if applicable), and column-type validation once, rather than many times.
+                ReadOnlyMemory<char> textValue = default;
+                VBuffer<ReadOnlyMemory<char>> tokensValue = default;
+
+                var textGetter = cursor.GetGetter<ReadOnlyMemory<char>>(textColumn);
+                var tokensGetter = cursor.GetGetter<VBuffer<ReadOnlyMemory<char>>>(tokensColumn);
+
+                while (cursor.MoveNext())
+                {
+                    textGetter(ref textValue);
+                    tokensGetter(ref tokensValue);
+
+                    Console.WriteLine($"{textValue} => {string.Join(", ", tokensValue.DenseValues())}");
+                }
+
+                // The output to console is this:
+
+                // Hello my friend. => Hello, my, friend.
+                // Stay awhile and listen. => Stay, awhile, and, listen.
+                // Masterfully done hero! => Masterfully, done, hero!
+
+                // Note that it may be interesting to set a breakpoint on the Console.WriteLine, and explore
+                // what is going on with the cursor, and the buffers. In particular, on the third iteration,
+                // while `tokensValue` is logically presented as a three element array, internally you will
+                // see that the arrays internal to that structure have (at least) four items, specifically:
+                // `Masterfully`, `done`, `hero!`, `listen.`. In this way we see a simple example of the details
+                // of how buffer sharing from one iteration to the next actually works.
+            }
+        }
+
+        private sealed class InputObject
+        {
+            public bool Label { get; }
+            public string Text { get; }
+
+            public InputObject(bool label, string text)
+            {
+                Label = label;
+                Text = text;
+            }
+        }
+
+        /// <summary>
+        /// This is an implementation of <see cref="IDataView"/> that wraps an <see cref="IEnumerable{T}"/>
```

Comment:
In my first attempt at this, I had written much of this material as separate XML docs for each member, but I found that this made it hard to tell a connected "story" about what is going on here, so I stuck to one big block on the `IDataView` implementation and one block on the `DataViewRowCursor` derived class. This made it easier to tell a cohesive story, while also having it I think look a lot less busy.

They are XML comments so I can use `<see` tags, mostly so that someone inspecting this sample can sometimes F12 to see what I'm talking about...

Reply:
A few points:

1) The docs engine will paste this examples verbatim, so none of the xml docstrings will be parsed and the F12 for cref won't work as you hoped them to.

2) If there's a way to put these in the <remarks> section for IDataView we can get all the xml features like cref. But you're writing a story that's attached to specific code blocks, so lumping it all in <remarks> would probably break the story.

3) This is ideal to be written as mixed xml + code pages like below but those pages are editorials on a different repo and are manually updated.
[https://docs.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/inspect-intermediate-data-ml-net](https://docs.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/inspect-intermediate-data-ml-net)

Having said all that, I think this works for V1 as is, with the only caveat the xml features like cref won't work.


---
In reply to: [274705835](https://github.com/dotnet/machinelearning/pull/3302#discussion_r274705835) [](ancestors = 274705835)

Reply:
Very good points @shmoradims. Thanks for keeping track of this. Fair enough. We can think about that later. TBH I find the documentation system perhaps more confusing than other parts of our infrastructure, so I'm grateful to have your guidance in this area.

I'm still not sure why the `<see` tags wouldn't work if someone were to try to run the sample in VS -- certainly we reference types outside of our own library all the time, e.g., lots of things in the BCL -- but we can try as you say to work that out later.